File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n=s.size();
        set<char> st;
        int i=0;
        int j=0;
        int ans=0;
        while(j<n){
            if(st.find(s[j])!=st.end()){
                while(s[i]!=s[j]){
                    st.erase(s[i]);
                    i++;
                }
                st.erase(s[i]);
                i++;
            }
            st.insert(s[j]);
            j++;

            ans=max(ans,(int)st.size());
        }
        return ans;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        char[] str=s.toCharArray();
        int n=str.length;

        Set<Character> mp=new HashSet<>();
        int i=0;
        int j=0;

        int ans=0;
        
        while(j<n){
            if(mp.contains(str[j])){
                while(str[i]!=str[j]){
                    mp.remove(str[i]);
                    i++;
                }
                mp.remove(str[i]);
                i++;
            }
            ans=Math.max(ans,j-i+1);
            mp.add(str[j]);
            j++;
        }
        return ans;
    }
}